File: Contests/CSEC/CSEC Contests -4th-Year/D_Gold_Rush.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool rec(long long n, long long m){
    if (n == m){
        return true;
    }
    if (n < m){
        return false;
    }
    if ((n%3 == 0) || (((2*n)%3) == 0)){
        if (m == (2 * (n / 3)) || m == (n / 3)){
            return true;
        }
        return rec((n/3), m) || rec(2*(n/3), m);
    }
    return false;
}


int main(){
    int t;

    cin >> t;

    for (int i = 0; i < t; i++)
    {
        long long n, m;
        cin >> n >> m;
        
        if (rec(n, m)){
            cout << "YES" << endl;
        }
        else{
            cout << "NO" << endl;
        }
    }
    

    return 0;
}

File: Contests/CSEC/CSEC Contests -4th-Year/D_Gold_Rush.py
import sys, threading

input = lambda: sys.stdin.readline().strip()

def main():
    m = 0
    def rec(n):
        nonlocal m
        if n == m:
            return True
        if n < m or n % 3:
            return False
        
        return rec(n // 3) or rec(n - (n//3))
        
    for i in range(int(input())):
        n , m = [int(i) for i in input().split()]

        if rec(n):
            print("YES")
        else:
            print("NO")
    
if __name__ == '__main__':
    
    sys.setrecursionlimit(1 << 30)
    threading.stack_size(1 << 27)

    main_thread = threading.Thread(target=main)
    main_thread.start()
    main_thread.join()

